pkg/k8s/client.go

```diff
@@ -271,101 +232,29 @@ func (c *Client) UpdateUpstreamTrafficSettingStatus(obj *policyv1alpha1.Upstream
 	return c.policyClient.PolicyV1alpha1().UpstreamTrafficSettings(obj.Namespace).UpdateStatus(context.Background(), obj, metav1.UpdateOptions{})
 }
 
-// ServiceToMeshServices translates a k8s service with one or more ports to one or more
-// MeshService objects per port.
-func (c *Client) ServiceToMeshServices(svc corev1.Service) []service.MeshService {
-	var meshServices []service.MeshService
-
-	for _, portSpec := range svc.Spec.Ports {
-		meshSvc := service.MeshService{
-			Namespace: svc.Namespace,
-			Name:      svc.Name,
-			Port:      uint16(portSpec.Port),
-		}
-
-		// attempt to parse protocol from port name
-		// Order of Preference is:
-		// 1. port.appProtocol field
-		// 2. protocol prefixed to port name (e.g. tcp-my-port)
-		// 3. default to http
-		protocol := constants.ProtocolHTTP
-		for _, p := range constants.SupportedProtocolsInMesh {
-			if strings.HasPrefix(portSpec.Name, p+"-") {
-				protocol = p
-				break
-			}
-		}
-
-		// use port.appProtocol if specified, else use port protocol
-		meshSvc.Protocol = pointer.StringDeref(portSpec.AppProtocol, protocol)
-
-		// The endpoints for the kubernetes service carry information that allows
-		// us to retrieve the TargetPort for the MeshService.
-		endpoints, _ := c.GetEndpoints(meshSvc)
-		if endpoints != nil {
-			meshSvc.TargetPort = GetTargetPortFromEndpoints(portSpec.Name, *endpoints)
-		} else {
-			log.Warn().Msgf("k8s service %s/%s does not have endpoints but is being represented as a MeshService", svc.Namespace, svc.Name)
-		}
-
-		if !IsHeadlessService(svc) || endpoints == nil {
-			meshServices = append(meshServices, meshSvc)
-			continue
-		}
+// IsHeadlessService determines whether or not a corev1.Service is a headless service
+func IsHeadlessService(svc corev1.Service) bool {
+	return len(svc.Spec.ClusterIP) == 0 || svc.Spec.ClusterIP == corev1.ClusterIPNone
+}
 
-		for _, subset := range endpoints.Subsets {
-			for _, address := range subset.Addresses {
-				if address.Hostname == "" {
-					continue
-				}
-				meshServices = append(meshServices, service.MeshService{
-					Namespace:  svc.Namespace,
-					Name:       svc.Name,
-					Subdomain:  address.Hostname,
-					Port:       meshSvc.Port,
-					TargetPort: meshSvc.TargetPort,
-					Protocol:   meshSvc.Protocol,
-				})
-			}
-		}
+// GetMeshConfig returns the current MeshConfig
+func (c *Client) GetMeshConfig() configv1alpha2.MeshConfig {
```

Comment:
Is this function in MeshCatalog and the compute client?

Reply:
nope it's just here. this gets embedded in the meshcatalog though so it has access to it.